src/controller/tests/TestSetUpCodePairer.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+ *
+ *    Copyright (c) 2020-2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <pw_unit_test/framework.h>
+
+#include <controller/SetUpCodePairer.h>
+#include <setup_payload/Base38Decode.h>
+#include <setup_payload/Base38Encode.h>
+
+#include <lib/core/StringBuilderAdapters.h>
+#include <lib/support/CHIPMemString.h>
+#include <lib/support/PersistentStorageMacros.h>
+#include <lib/support/Span.h>
+#include <lib/support/TestPersistentStorageDelegate.h>
+
+#include <cstring>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+using namespace chip;
+using namespace chip::Controller;
+
+namespace {
+
+std::string to_hex_string(uint8_t value)
+{
+    std::stringstream ss;
+    ss << std::hex << std::setw(2) << std::setfill('0') << static_cast<int>(value);
+    return ss.str();
+}
+
+class SetUpCodePairerTest : public ::testing::Test
+{
+public:
+    static void SetUpTestSuite() { ASSERT_EQ(chip::Platform::MemoryInit(), CHIP_NO_ERROR); }
+    static void TearDownTestSuite() { chip::Platform::MemoryShutdown(); }
+
+protected:
+};
+
+TEST_F(SetUpCodePairerTest, SuccessfulyPopulatesPayloadFromQRSetUpCode)
```

Comment:
This should all be covered by existing setup_payload unit tests, including the ones added in #38728.  See `src/setup_payload/tests/TestSetupPayload.cpp`.